xbmc/dialogs/GUIDialogSelect.h

```diff
@@ -55,8 +57,11 @@ class CGUIDialogSelect : public CGUIDialogBoxBase
 
 private:
   bool m_bButtonEnabled;
+  bool m_bButton2Enabled;
   bool m_bButtonPressed;
+  bool m_bButton2Pressed;
   int m_buttonLabel;
+  int m_button2Label;
```

Comment:
About here to use the integer I'm not sure the best :thinking:.
Before about OK and Cancel it can stays the same and with integer. But with third I think the name not always the same.

If now a addon (python/binary) does use it (still need then changes inside his API about new button) is it forced to use strings from Kodi itself.

Maybe it makes then sense to use as "string" direct?
e.g.:
```cpp
  void EnableButton(bool enable, int label);
  void EnableButton(bool enable, const std::string& label);
  void EnableButton2(bool enable, int label);
  void EnableButton2(bool enable, const std::string& label);

  ...

  std::string m_buttonLabel;
  std::string m_button2Label;
```

```cpp
void CGUIDialogSelect::EnableButton(bool enable, int label)
{
  m_bButtonEnabled = enable;
  m_buttonLabel = g_localizeStrings.Get(label);
}

void CGUIDialogSelect::EnableButton(bool enable, const std::string& label)
{
  m_bButtonEnabled = enable;
  m_buttonLabel = label;
}

void CGUIDialogSelect::EnableButton2(bool enable, int label)
{
  m_bButton2Enabled = enable;
  m_button2Label = g_localizeStrings.Get(label);
}

void CGUIDialogSelect::EnableButton2(bool enable, const std::string& label)
{
  m_bButton2Enabled = enable;
  m_button2Label = label;
}
```

Think features like them are mainly taken from addons.

Reply:
Thanks @AlwinEsch I have incorporated holding button labels as string not int, good suggestion for possible future addon use.